clang-tools-extra/clangd/unittests/IncludeCleanerTests.cpp

```diff
@@ -135,6 +135,21 @@ TEST(IncludeCleaner, GetUnusedHeaders) {
                            Pointee(writtenInclusion("\"dir/unused.h\""))));
 }
 
+TEST(IncludeCleaner, SystemUnusedHeaders) {
+  auto TU = TestTU::withCode(R"cpp(
+    #include <system_header.h>
+    #include <system_unused.h>
+    SystemClass x;
+  )cpp");
+  TU.AdditionalFiles["system/system_header.h"] = guard("class SystemClass {};");
+  TU.AdditionalFiles["system/system_unused.h"] = guard("");
+  TU.ExtraArgs = {"-isystem", testPath("system")};
```

Comment:
using `-I` instead of `-isystem` might be better here. as in the implementation we don't really disambiguate between type of the inclusions, just how they're spelled.

Reply:
I've copied `-isystem` from the `GetUnusedHeaders` test.
If `AnalyzeSystemHeaders` / `AnalyzeAngledIncludes` must work for inclusions of any type, then maybe it'd be better to test both `-I` and `-isystem`?

Reply:
thanks, sgtm